api/python/provisioner/utils.py

```diff
@@ -218,3 +222,37 @@ def repo_tgz(
     run_subprocess_cmd(cmd)
 
     return dest
+
+
+# Validates the hostname string in config.ini
+# against hostname in CLI args
+def node_hostname_validator(
+    nodes,
+    config_path
+):
+    node_dict = {}
+    for node in nodes:
+        node_dict[node.minion_id] = node.host
+
+    logger.debug(
+        "Validating list of nodes: "
+        f"{yaml.dump(node_dict, default_flow_style=False)}"
+    )
+    logger.debug(f"Config file path: {config_path}")
+
+    parser_obj = configparser.ConfigParser()
+    parser_obj.read(config_path)
+
+    for section in parser_obj.sections():
+        if ("srvnode" in section
+            and
+            (
+                node_dict[parser_obj[section]]
```

Comment:
`node_dict[parser_obj[section]] ` In this section will give value as srvnode-1/srvndoe-2
so we need to make change:
`node_dict[section]`